doc/source/rayserve/overview.rst

```diff
@@ -0,0 +1,197 @@
+  
+  def handle_request(flask_request):
+    return "hello world"
+
+  class RequestHandler:
+    def __init__(self):
+        self.msg = "hello, world!"
+
+    def __call__(self, flask_request):
+        return self.msg
+
+  serve.create_backend(handle_request, "my_endpoint_backend")
+  serve.create_backend(RequestHandler, "my_endpoint_backend_class")
+
+Lastly, we need to link the particular backend to the server endpoint. 
+To do that we'll use the ``link`` capability.
+A link is essentially a load-balancer and allow you to define queuing policies 
+for how you would like backends to be served via an endpoint.
+For instance, you can route 50% of traffic to Model A and 50% of traffic to Model B.
+
+.. code-block:: python
+
+  serve.link("my_endpoint_backend", "my_endpoint")
+
+Once we've done that,we can now query our endpoint with something like requests 
+(or an external service).
+
+.. code-block:: python
+  
+  import requests
+  print(requests.get("http://127.0.0.1:8000/-/routes", timeout=0.5).text)
+
+Configuring Backends
+~~~~~~~~~~~~~~~~~~~~
+
+There are a number of things you'll likely want to do with your serving application including
+scaling out, splitting traffic, or batching input for better response performance. To do all of this,
+you will create a ``BackendConfig``, a configuration object that you'll use to set 
+the properties of a particular backend. That will allow you to do things like scale out,
+split traffic, and batch requests for better performance.
+
+Scaling Out
+~~~~~~~~~~~
+
+To scale out a backend to multiple workers, simplify configure the number of replicas.
+
+.. code-block:: python
+
+  config = serve.BackendConfig(num_replicas=2)
+  serve.create_backend(handle_request, "my_scaled_endpoint_backend", backend_config=config)
+
+This will scale out the number of workers that can accept requests.
+
+Splitting Traffic
+~~~~~~~~~~~~~~~~~
+
+It's trivial to also split traffic, simply specify the endpoint and the backends that you want to split.
+
+.. code-block:: python
+  
+  serve.create_endpoint("endpoint_identifier_split", "/split", methods=["GET", "POST"])
+
+  # splitting traffic 70/30
+  serve.split("endpoint_identifier_split", {"my_endpoint_backend": 0.7, "my_endpoint_backend_class": 0.3})
+
+
+Batching
+~~~~~~~~
+
+You can also have RayServe batch requests for performance. You'll configure this in the backend config.
+
+.. code-block:: python
+
+  class BatchingExample:
+      def __init__(self):
+          self.count = 0
+
+      @serve.accept_batch
+      def __call__(self, flask_request):
+          self.count += 1
+          batch_size = serve.context.batch_size
+          return [self.count] * batch_size
+
+  serve.create_endpoint("counter1", "/increment")
+
+  config = BackendConfig(max_batch_size=5)
+  serve.create_backend(
+      BatchingExample, "counter:v11", backend_config=config)
+  serve.link("counter1", "counter:v11")
+
+Other Resources
+----------------
+
+More coming soon!
```

Comment:
@simon-mo can you add the automodules here? We had them in a previous version

Reply:
@anabranch please bring the runnable examples back.

Reply:
```rst
Serve a stateless function:

.. literalinclude:: ../../../python/ray/serve/examples/doc/quickstart_function.py

Serve a stateful class:

.. literalinclude:: ../../../python/ray/serve/examples/doc/quickstart_class.py
```